producer.go

```diff
@@ -233,7 +268,13 @@ func (p *Producer) Destroy() error {
 	if err := p.conn.removeSchemaUpdatesListener(p.stationName); err != nil {
 		return memphisError(err)
 	}
-	return p.conn.destroy(p)
+	err := p.conn.destroy(p)
+	if err != nil {
+		return err
+	}
+
+	checkForProducerExistenceAndDelete(p)
```

Comment:
you should add it to the the station destruction function and to the memphis.close as well